django/src/rdwatch/views/site_image.py

```diff
@@ -146,3 +151,71 @@ def site_images(request: HttpRequest, id: UUID4):
     output['evaluationGeoJSON'] = site_eval_data['json']['evaluationGeoJSON']
     output['evaluationBBox'] = site_eval_data['json']['evaluationBBox']
     return output
+
+
+@router.post('/{id}/image_embedding', response=UUID4)
+def post_image_embedding(request: HttpRequest, id: int):
+    if not SiteImage.objects.filter(pk=id).exists():
+        raise Http404()
+    else:
+        return generate_image_embedding.delay(id).id
+
+
+@router.get('/{id}/image_embedding_status/{uuid}')
+def get_image_embedding_status(request: HttpRequest, id: int, uuid: UUID4):
+    if not SiteImage.objects.filter(pk=id).exists():
+        raise Http404()
+    else:
+        task = AsyncResult(uuid)
+        result = {
+            'state': task.state,
+            'status': task.status,
+        }
+        return result
+
+
+@router.get('/{id}/image', response=SiteImageSchema)
+def get_site_image(request: HttpRequest, id: int):
+    try:
+        site_image = (
+            SiteImage.objects.filter(pk=id)
+            .annotate(
+                timestamp_epoch=ExtractEpoch('timestamp'),
+                bbox=BoundingBox('image_bbox'),
+            )
+            .values(
+                'timestamp_epoch',
+                'source',
+                'cloudcover',
+                'image',
+                'observation_id',
+                'bbox',
+                'percent_black',
+                'image_dimensions',
+                'aws_location',
+                'image_embedding',
+                'pk',  # Assuming 'pk' is the primary key field name
+            )
+            .first()
+        )
+    except SiteImage.DoesNotExist:
+        raise Http404()
+
+    if site_image:
+        response_data = {
+            'timestamp': site_image['timestamp_epoch'],
+            'source': site_image['source'],
+            'cloudcover': site_image['cloudcover'],
+            'image': default_storage.url(site_image['image']),
+            'observation_id': site_image['observation_id'],
+            'percent_black': site_image['percent_black'],
+            'bbox': site_image['bbox'],
+            'image_dimensions': site_image['image_dimensions'],
+            'aws_location': site_image['aws_location'],
+            'image_embedding': default_storage.url(site_image['image_embedding']),
+            'id': site_image['pk'],
+        }
```

Comment:
I think we should consider aliasing these in the `SiteImageSchema`, that there's no need to repeat all the values here and you can just do `return site_image` directly.

Reply:
We can do most of this, except I think because we are using an older ninja version we might not be able to use resolvers for converting the image/image_embedding base URLS into resigned URLs.  Or there may be a way to do it and I just don't see it in the documentation.

Reply:
Yeah, I'm not sure either. I'm fine waiting until we upgrade ninja to address this